crates/client/db/src/lib.rs

```diff
@@ -175,33 +280,47 @@ impl DeoxysBackend {
                         paritydb_path: starknet_database_dir(db_config_dir, "paritydb"),
                         cache_size: 0,
                     },
-                    _ => return Err("Supported db sources: `rocksdb` | `paritydb` | `auto`".to_string()),
+                    _ => bail!("Supported db sources: `rocksdb` | `paritydb` | `auto`"),
```

Comment:
This is more out of curiosity: when do you consider it to be a better fit to use `thiserror` vs `anyhow`?

Reply:
My thinking was: thiserror for everything, anyhow for stuff that sets up the startup.
However, I think it's kinda bad for user experience as anyhow's messages are more developer-oriented and show stacktraces vs errors that are user errors
maybe we should just put everything on thiserror at some point